scripts/deploy.js

```diff
@@ -138,10 +138,53 @@ async function deployAll(
     };
 }
 
+async function deployAllWithTestGatewayCaller(
+    wallet,
+    chainName,
+    evmChains = [],
+    deploymentKey = 'InterchainTokenService',
+    factoryDeploymentKey = deploymentKey + 'Factory',
+) {
+    const create3Deployer = await new ethers.ContractFactory(Create3Deployer.abi, Create3Deployer.bytecode, wallet)
+        .deploy()
+        .then((d) => d.deployed());
+    const gateway = await deployMockGateway(wallet);
+    const gasService = await deployGasService(wallet);
+
+    const interchainTokenServiceAddress = await getCreate3Address(create3Deployer.address, wallet, deploymentKey);
+    const tokenManagerDeployer = await deployContract(wallet, 'TokenManagerDeployer', []);
+    const interchainToken = await deployContract(wallet, 'InterchainToken', [interchainTokenServiceAddress]);
+    const interchainTokenDeployer = await deployContract(wallet, 'InterchainTokenDeployer', [interchainToken.address]);
+    const tokenManager = await deployContract(wallet, 'TokenManager', [interchainTokenServiceAddress]);
+    const tokenHandler = await deployContract(wallet, 'TokenHandler', [gateway.address]);
+    const gatewayCaller = await deployContract(wallet, 'TestGatewayCaller');
```

Comment:
```suggestion
    const gatewayCaller = await deployContract(wallet, contractOverrides.GatewayCaller || 'GatewayCaller');
```
This can optionally overriden by the caller. You can add a docstring on the method that this is useful for mocking contracts in tests

Reply:
test code shouldn't be added to this folder. I prefer adding a proper deployment fix similar to my suggestion above instead of a custom test deployment method that we now have to maintain. Or, do the custom deployment within the test itself